src/server/services/cache/redis.ts

```diff
@@ -0,0 +1,95 @@
+import Redis from 'ioredis'
+
+import { Cache, Key, Value, ValueCallback } from './types'
+import {DEFAULT_CACHE_TTL} from "../../../config";
+
+export const RedisCache = (): Cache => {
+  const redis = new Redis(process.env.REDIS_URL as string)
+    redis.on('connect', () => {
+        console.log('CacheStore - Connection status: connected');
+    });
+    redis.on('end', () => {
+        console.log('CacheStore - Connection status: disconnected');
+    });
+    redis.on('reconnecting', () => {
+        console.log('CacheStore - Connection status: reconnecting');
+    });
+    redis.on('error', (err) => {
+        console.log('CacheStore - Connection status: error ', { err });
+    });
+  return {
+    getAndSet<T extends Value>(
+      key: Key,
+      valueFunction: ValueCallback<T>,
+      ttl: number = DEFAULT_CACHE_TTL,
+    ): Promise<T> {
+      return new Promise<T>((resolve) => {
```

Comment:
Mit `this.get` hast du bereits eine promise, somit musst du nicht nochmals eine erstellen


```
        return this.get(key).then((value) => {
          if (value) {
            return (value as T)
          } else {
            return valueFunction().then((newValue) => {
              this.set(key, newValue, ttl)
              return newValue;
            })
          }
        })
```

Reply:
Merci, ist verbessert